### PR TITLE
refactor: Converts Devise to API only mode

### LINKED ISSUE #746: Convert devise to API only mode

### PR DESCRIPTION
**Pull request by JeremyRudman**
**Summary** 
This refactors the authentication system to use JWT tokens in order to complete any task. This works by having a user register a account and then login. The login route will provide the user with a JWT token they can use to access all routes routes in the rails application (although they will need the correct role to do anything in those routes).

Note that this breaks the doorkeep authentication for third party applications

**Test**
All the Unit tests now require JWT tokens to pass and if the lack them the task will always return unauthorized.

Using postman I registered a account and logged to obtain a JWT token.
With that JWT token I was successfully able to get the bus route for the specific user the JWT token is associated with.

### CODE REVIEW

**Comment by JeremyRudman** [comment]
@cbaudouinjr This isn't done yet but its mostly finished and I am not sure how long the last 5% will take. It will take a while to review so why don't you start reviewing now while I figure out myMLH integration.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,12 +1,49 @@
 class UsersController < ApplicationController
-  before_action :logged_in
   respond_to :json
 
-  def logged_in
-    authenticate_user!
+  before_action :authenticate_user!, except: [:login, :register]
+  before_action :find_user
+  skip_before_action :authenticate_user, only: [:login, :register]
```
Comment (cbaudouinjr) [inline app/controllers/users_controller.rb:5]: May be redundant.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,12 +1,49 @@
...
+
+  def login
+    user = User.find_by_email(params[:email])
+
+    if user&.valid_password?(params[:password])
+      @current_user = user
+      render json: user.generate_jwt
+    else
+      render json: { errors: { 'email or password' => ['is invalid'] } }, status: :unprocessable_entity
```
Comment (cbaudouinjr) [inline app/controllers/users_controller.rb]: Can we make this an ErrorResponse.rb object?

File: test/test_helper.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@
 require "webmock/minitest"
 require_relative './matchers'
 
-Minitest::Reporters.use!
+Minitest::Reporters.use! unless ENV['RM_INFO']
```
Comment (cbaudouinjr) [inline test/test_helper.rb:29]: What is this?

**Review by cbaudouinjr — CHANGES_REQUESTED** [review]

File: test/test_helper.rb (hunk 3 above)
Comment (JeremyRudman) [reply test/test_helper.rb:29]: I followed jet brains article to get there debugger to work

https://blog.jetbrains.com/ruby/2021/04/improved-minitest-support-action-required/

**Pull request closed without merging**